site/content/blog/2022-08-25-zap-on-raspberry-pi/index.md

```diff
@@ -0,0 +1,140 @@
+
+Now all we have to do is unplug the SD card from the card reader and insert into the raspberry pi and connect it to a power source and wait for the booting process to finish up.
+
+## Step 2 : Connecting your raspberry pi to your android phone
+
+Next we turn on our router or mobile hotspot with the wifi setup preinstalled in Step1 to enable raspberry pi connect automatically to it and create a network in which both devices can share connectivity
+
+Using any ssh client , connect to the ip address(192.168.43.112 in my case) of raspberry pi on the network using the preconfigured ssh credentials in Step 1 to enable access to a terminal window
+   
+![Neofetch ssh screen](images/IMG_20220819_105041.jpg)
+
+Run the following commands to install a VNC server on the raspberry pi
+
+* Update the raspberry pi repository list
+
+```
+ sudo apt update   
+```
+
+* Install tightvncserver by typing
+```
+  sudo apt install tightvncserver
+```
+
+* Run tightvncserver
+```
+    sudo tightvncserver
+```
+
+Now create a password for the VNC server .There's no need to create a view only password 
+
+The Raspberry Pi has now started a VNC server session at ***your raspberry pi ip:1*** remember this address as it is needed later.
+
+Install any VNC client software from the Google Playstore and launch the application add new host using ***your raspberry pi ip:1*** as Host address, use the user that created the vnc server on your raspberry pi and the password you created.
+
+## Step 3 : Installing a windows manager and a desktop manager on the raspberry pi
+When i was going through this set up initially i tried to make it as light weight and minimalistic as possible, i installed only X , dmenu and a windows manager , i3  but the VNC clients weren't recognising the shortcut keys to run the terminal and dmenu  even after editing i3's config so i had to install a desktop manager , lxde.
+  
+![i3 default display](images/Screenshot_2022-08-19-18-10-51-590_com.realvnc.viewer.android.jpg)
+
+Run the following commands to install X, dmenu ,i3 and lxde on the raspberry pi
+
+```
+sudo apt update && sudo apt upgrade && sudo apt install xserver-xorg-core xserver-xorg-video-fbdev xserver-xorg-input-evdev xinit xfonts-base lxde  --no-install-recommends -y
+```
+
+Let XServer know the preferred desktop environment  to start:
+
+```
+echo 'exec startlxde' > ~/.xinitrc
+```
+
+Now we can start up a terminal from the VNC client by clicking the last option on the lower left of the screen and selecting system tools
+
+![raspberry pi vnc terminal screen](images/Screenshot_2022-08-20-09-17-08-469_com.realvnc.viewer.android.jpg)
+
+## Step 4: Installing ZAP on the raspberry pi
+Before we start ZAP installation, we need to install java8 or higher in our raspberry pi (i used java11 for this installation)
+
+```
+sudo apt-get install openjdk-8-jdk
+```
+
+Go to [ZAP official download page](https://www.zaproxy.org/download/) and download the ***`core cross platform package`*** zip file.
```

Comment:
As its the same site we can replace `https://www.zaproxy.org/download/` with `/download/`